account_payment_cc/account_voucher.py

```diff
@@ -0,0 +1,236 @@
+﻿# -*- coding: utf-8 -*-
+
+from openerp.osv import orm
+
+
+class account_voucher(orm.Model):
+
+    _inherit = 'account.voucher'
+
+    def action_move_line_create(self, cr, uid, ids, context=None):
+        base_func = super(account_voucher, self).action_move_line_create
+        res = base_func(cr, uid, ids, context)
+        move_line_pool = self.pool.get('account.move.line')
+        move_pool = self.pool.get('account.move')
+
+        for voucher in self.browse(cr, uid, ids):
+            if not voucher.journal_id.support_creditcard_transactions:
+                break
+
+            company_currency = self._get_company_currency(cr, uid, voucher.id,
+                                                          context)
+            current_currency = self._get_current_currency(cr, uid, voucher.id,
+                                                          context)
+
+            partner_id = voucher.journal_id.partner_id
+
+            account_payable = partner_id.property_account_payable
+            account_receivable = partner_id.property_account_receivable
+            account = voucher.journal_id.default_credit_account_id
+
+            if voucher.type in ('receipt', 'sale'):
+                account = voucher.journal_id.default_debit_account_id
+
+            # Create the account move record.
+            account_move_id = self.account_move_get(cr, uid, voucher.id,
+                                                    context=context)
+            move_id = move_pool.create(cr, uid, account_move_id,
+                                       context=context)
+
+            fline = self.first_move_line_get(
+                cr, uid, voucher.id, move_id, company_currency,
+                current_currency, context
+            )
+
+            fline.update({
+                'partner_id': (
+                    partner_id and
+                    partner_id.id or
+                    voucher.partner_id.id
+                ),
+            })
+
+            credit, debit = fline.get('credit'), fline.get('debit')
+            alines = [line.id for line in voucher.line_ids if line.amount]
+            ctx = context and context.copy() or {}
+            ctx.update({'date': voucher.date})
+
+            if alines:
```

Comment:
This is the only place this variable is used. Why not use:

```
if any(line for line in voucher.line_ids if line.amount):
```

It could save some computation